lib/ansible/inventory/manager.py

```diff
@@ -360,8 +360,8 @@ def get_hosts(self, pattern="all", ignore_limits=False, ignore_restrictions=Fals
                 # mainly useful for hostvars[host] access
                 if not ignore_limits and self._subset:
                     # exclude hosts not in a subset, if defined
-                    subset = self._evaluate_patterns(self._subset)
-                    hosts = [h for h in hosts if h in subset]
+                    subset_uuids = [s._uuid for s in self._evaluate_patterns(self._subset)]
+                    hosts = [h for h in hosts if h._uuid in subset_uuids]
```

Comment:
unsure if this will survive a `meta: refresh_inventory` as the inventory hosts will get new uuids

Reply:
This shouldn't be a change in behavior.  In the end, it does the same thing as it did previously.  The `__eq__` check in `Host` compares the `_uuid`.  This just short circuits some things to make the comparison faster here.